### PR TITLE
RHOAIENG-3984 Use the proxy at Red Hat.

### PR DESCRIPTION
**Pull request by pilhuhn**
https://issues.redhat.com/browse/RHOAIENG-3984

## Description

Make segment connections go through the proxy at console.redhat.com

## How Has This Been Tested?

Look at Browsers network tab

## Test Impact

Should be none, as the tracking itself has not changed.

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/utilities/segmentIOUtils.tsx [hunk 1]
```diff
@@ -75,7 +75,7 @@ export const initSegment = async (props: {
       const key = analytics.methods[e];
       analytics[key] = analytics.factory(key);
     }
-    analytics.load = (key: string, e: Event) => {
+    analytics.load = (key: string, options: any) => {
```
Comment (andrewballantyne) [inline frontend/src/utilities/segmentIOUtils.tsx:78]: We have a no `any` policy. 

Looks like the type is https://github.com/segmentio/analytics-next/blob/master/packages/browser/src/core/analytics/index.ts#L86 -- maybe we type some of this internally at some point... but I think you just need to stop eslint from having an issue.

```suggestion
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    analytics.load = (key: string, options: any) => {
```

File: frontend/src/utilities/segmentIOUtils.tsx [hunk 2]
```diff
@@ -84,11 +84,19 @@ export const initSegment = async (props: {
...
-      analytics.load(segmentKey);
+      analytics.load(segmentKey, {
+        cdnURL: 'console.redhat.com/connections/cdn',
+        integrations: {
+          'Segment.io': {
+            apiHost: 'console.redhat.com/connections/api/v1',
+            protocol: 'https',
+          },
+        },
+      });
```
Comment (andrewballantyne) [inline frontend/src/utilities/segmentIOUtils.tsx:99]: @pilhuhn is there anyway I can test this? Or see the backend for requests making it there?

**Comment by andrewballantyne** [comment]
@pilhuhn can you rebase and squash your commits -- we don't typically want merge commits in our PRs.

File: frontend/src/utilities/segmentIOUtils.tsx (hunk 2 above)
Comment (pilhuhn) [reply frontend/src/utilities/segmentIOUtils.tsx:100]: Yes. In the browser network tab.
Without the change:
Bildschirmfoto 2024-04-03 um 16 25 55
with the change:
Bildschirmfoto 2024-04-03 um 16 21 29

Note, in dev-mode no analytics is sent out, so those traces would not be in the network tab, despite the console talking about such events.

**Comment by pilhuhn** [comment]
@andrewballantyne Yeah. This is because I accepted your suggestion from the github UI :)

**Review by andrewballantyne — APPROVED** [review]

**Comment by andrewballantyne** [comment]
/retest

Flake?

```
[1]   ClusterStorage
[1]     1) Empty state
[1]     ✓ Add cluster storage (2140ms)
[1]     ✓ list accelerator profiles and Table sorting (1048ms)
[1]     ✓ Edit cluster storage (1133ms)
[1]     ✓ Delete cluster storage (1104ms)
[1] 
[1] 
[1]   4 passing (7s)
[1]   1 failing
[1] 
[1]   1) ClusterStorage
[1]        Empty state:
[1]      Error: Failed to visit '/projects/test-project?section=cluster-storages'. Status code: unknown
[1]       at Context.eval (webpack:///./cypress/support/commands/application.ts:20:0)
```

**Comment by shalberd** [comment]
oh man, I hope that with "Make segment connections go through the proxy at console.redhat.com" you have not again introduced a hardcoded thing that is incompatible with corporate setups.

**Comment by shalberd** [comment]
This console.redhat.com is not a proxy, it is a telemetry tool ... this will not work in any corporate environment where direct access to console.redhat.com is not allowed, only thru proxy defined at central cluster level https://docs.openshift.com/container-platform/4.14/networking/enable-cluster-wide-proxy.html#nw-proxy-configure-object_config-cluster-wide-proxy  and by extension envs HTTP_PROXY and HTTPS_PROXY pointing to that central cluster config proxy.

**Pull request merged**